sourced/ml/cmd_entries/repos2bow.py

```diff
@@ -66,12 +66,15 @@ def repos2bow_entry_template(args, select=HeadFiles, cache_hook=None):
         .prune(args.min_docfreq) \
         .greatest(args.vocabulary_size) \
         .save(args.docfreq)
-    uast_extractor \
+    bags_writer= uast_extractor \
         .link(BagFeatures2TermFreq()) \
         .link(TFIDF(df_model)) \
         .link(document_indexer) \
-        .link(Indexer(Uast2BagFeatures.Columns.token, df_model.order)) \
-        .link(BOWWriter(document_indexer, df_model, args.bow, args.batch)) \
+        .link(Indexer(Uast2BagFeatures.Columns.token, df_model.order))\
+        .link(Cacher.maybe(args.persist))
```

Comment:
I have a feeling that `save_hook` should allow to change BOWWriter to an arbitrary transformer. Thus you do not need to cache. I don't know why it is needed upstream; another option is to filter the data through save_hook() and link everything together without a fork.